cvmfs/sync_union_tarball.cc

```diff
@@ -233,6 +234,10 @@ void SyncUnionTarball::ProcessArchiveEntry(struct archive_entry *entry) {
   } else {
     LogCvmfs(kLogUnionFs, kLogStderr,
              "Fatal error found unexpected file: \n%s\n", filename.c_str());
+    // if for any reason this code path change and we don't abort anymore,
+    // remember to wakeup the signal, otherwise we will be stuck in a deadlock
+    //
+    // read_archive_signal_->Wakeup();
```

Comment:
@jblomer  do you think this is better?

I left the Wakeup like there from a previous iteration where I spend too much time figuring out why it was deadlocking.

Maybe I should just leave the comment and remove the actual code line?

Reply:
Oh, I see.  Yep, that's fine.